File: 69.sqrt-x.js
/*
 * @lc app=leetcode id=69 lang=javascript
 *
 * [69] Sqrt(x)
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    var l=0;
    var r = x;
    while(l<=r){
        let mid = l + Math.floor((r-l)/2);
        //console.log(mid);
        if(mid * mid === x){
            return mid;
        }else if(mid * mid>x){
            r = mid -1;
        }else{
            l = mid +1 ;
        }
    }
    //console.log(l + " "+ r);
    return r;
    
};
// @lc code=end

File: 69.sqrt-x.java
/*
 * @lc app=leetcode id=69 lang=java
 *
 * [69] Sqrt(x)
 */

// @lc code=start
class Solution {
    public int mySqrt(int x) {
        if (x == 0) {
            return 0;
        }
        if (x == 1) {
            return 1;
        }
        int left = 1;
        int right = x;
        long num;
        while (left < right) {
            int mid = left + (right - left) / 2;
            num = (long)mid * mid;    
            if (num > x) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        return left - 1;
        
    }
}
// @lc code=end

